cruise-control/src/yaml/endpoints/rightsize.yaml

```diff
@@ -0,0 +1,89 @@
+RightsizeEndpoint:
+  post:
+    operationId: rightsize
+    summary: Test the provisioner on Azure
+    parameters:
+      - name: num_brokers_to_add
+        in: query
+        description: The difference in broker count to rightsize towards.
+        schema:
+          type: integer
+          format: int32
+          default: -1
+      - name: partition_count
+        in: query
+        description: The target number of partitions to rightsize towards.
+        schema:
+          type: integer
+          format: int32
+          default: -1
+      - name: topic
+        description: Regular expression to specify subject topics.
+        schema:
+          type: string # topics regex
+          default: null
+          example: topic_%5B0-9%5D.%2A
+      - name: allow_capacity_estimation
+        in: query
+        description: Whether to allow capacity estimation when cruise-control is unable to obtain all per-broker capacity information.
+        schema:
+          type: boolean
+          default: true
+      - name: doAs
+        in: query
+        description: The user specified by a trusted proxy in that authentication model.
+        schema:
+          type: string
+      - name: exclude_recently_demoted_brokers
+        in: query
+        description: Whether to allow leader replicas to be moved to recently demoted brokers.
+        schema:
+          type: boolean
+          default: false
+      - name: get_response_schema
+        in: query
+        description: Whether to return JSON schema in response header or not.
+        schema:
+          type: boolean
+          default: false
+      - name: json
+        in: query
+        description: Whether to return in JSON format or not.
+        schema:
+          type: boolean
+          default: false
+      - name: verbose
+        in: query
+        description: Return detailed state information.
+        schema:
+          type: boolean
+          default: false
+    responses:
+      '200':
+        description: Successful add brokers response.
+        content:
+          application/json:
+            schema:
+              $ref: '../responses/optimizationResult.yaml#/OptimizationResult'
+          text/plain:
+            schema:
+              type: string
+      '202':
```

Comment:
Note / reminder: the in progress response might disappear if we make this sync a request.